### PR TITLE
extralife overlay = no autoplay

### PR DESCRIPTION
**Pull request by chrisdugne**
User must click on overlay to activate video + extralife
==> overriding `customProps.autostart` video option if there is an overlay

**Testing Strategy**

- [x] Already covered by tests
- [x] Manual testing
- [x] Unit testing

### CODE REVIEW

**Comment by chrisdugne** [comment]
[extra-life: user must click on overlay to activate](https://trello.com/c/9MbwX3xi/1525-extra-life-user-must-click-on-overlay-to-activate)

**Comment by VincentCATILLON** [comment]
Can you add before/after gifs?

File: packages/@coorpacademy-components/src/molecule/resource-player/index.js [hunk 1]
```diff
@@ -135,7 +136,11 @@ class ResourcePlayer extends React.Component {
     return (
       <div data-name={type} className={className}>
         {overlayView}
-        <ResourceElement {...this.props} autoplay={this.state.autoplay} />
+        <ResourceElement
+          {...this.props}
+          autostart={!this.props.overlay}
+          autoplay={this.state.autoplay}
+        />
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-components/src/molecule/resource-player/index.js]: Missing prop type definition in this file

File: packages/@coorpacademy-components/src/molecule/resource-player/test/fixtures/jwplayer-with-overlay.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ export default {
         playerScript: JWPLAYER_SCRIPT_URL,
         licenseKey: 'QDh3Fb2afiIAFI+XwlncwQDhNEwkXetm1y8tzWn3km8=',
         customProps: {
-          autostart: false,
+          autostart: 'viewable',
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-components/src/molecule/resource-player/test/fixtures/jwplayer-with-overlay.js]: this prop should be bool, isnt it ?

File: packages/@coorpacademy-components/src/molecule/video-player/jwplayer.js [hunk 3]
```diff
@@ -142,6 +150,8 @@ class JWPlayer extends React.Component {
 }
 
 JWPlayer.propTypes = {
+  autostart: PropTypes.bool,
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-components/src/molecule/video-player/jwplayer.js]: I don't understand why there is too prop types `autostart`, can you explain please ?

File: packages/@coorpacademy-components/src/molecule/video-player/jwplayer.js [hunk 4]
```diff
@@ -154,6 +164,7 @@ JWPlayer.propTypes = {
           default: PropTypes.boolean
         })
       ),
+      autostart: PropTypes.oneOf(['viewable', 'false']),
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-components/src/molecule/video-player/jwplayer.js]: are you sure about it ? why not only boolean ?

Comment (chrisdugne) [reply packages/@coorpacademy-components/src/molecule/video-player/jwplayer.js]: from https://developer.jwplayer.com/jwplayer/docs/jw8-player-configuration-reference 

```
autostart: String
Whether the player will attempt to begin playback automatically when  a page is loaded. Set to 'viewable' to have player autostart if 50% is  viewable.
```

File: packages/@coorpacademy-components/src/molecule/video-player/jwplayer.js (hunk 3 above)
Comment (chrisdugne) [reply packages/@coorpacademy-components/src/molecule/video-player/jwplayer.js]: naming issue you're right this one should be
```
 disableAutostart: PropTypes.bool,
```

File: packages/@coorpacademy-components/src/molecule/resource-player/index.js (hunk 1 above)
Comment (chrisdugne) [reply packages/@coorpacademy-components/src/molecule/resource-player/index.js]: added within `ResourceElement.propTypes`

**Pull request merged**